pkg/reconciler/route/resources/ingress.go

```diff
@@ -115,6 +117,7 @@ func MakeIngressSpec(
 	tls []v1alpha1.IngressTLS,
 	clusterLocalServices sets.String,
 	targets map[string]traffic.RevisionTargets,
+	additionalRules ...v1alpha1.IngressRule,
```

Comment:
Instead of adding `additionRules` as the input, I would prefer passing `HTTP01Challenge` or `Certificate` as input, and building the IngressRule from HTTP01 Challenge in this function.
The reason is having `additionalRules` as input could make this function misused as we actually have implicit requirement for the`additionalRules` here, which is they will be put in the beginning of all of rules, as a result will have higher matching priority than other rules. We probably don't want to extend its usage elsewhere.